es6.md

```diff
@@ -189,3 +189,48 @@ interface ImportNamespaceSpecifier {
 ```
 
 A namespace import specifier, e.g., `* as foo` in `import * as foo from "mod.js"`.
+
+## ExportNamedDeclaration
+
+```js
+interface ExportNamedDeclaration <: Node {
+    declaration: Declaration | null;
+    specifiers: [ ExportSpecifier ];
+    source: Literal | null;
+}
+```
+
+An export named declaration, e.g., `export {foo, bar};`, `export {foo} from "mod";` or `export var foo = 1;`.
+
+_Note: Having `declaration` populated with non-empty `specifiers` or non-null `source` results in an invalid state._
+
+## ExportSpecifier
+
+```js
+interface ExportSpecifier {
+    exported: Identifier;
+    local: Identifier;
+}
+```
+
+An exported variable binding, e.g., `{foo}` in `export {foo}` or `{bar as foo}` in `export {bar as foo}`. The `exported` field refers to the name exported in the module. The `local` field refers to the binding into the local module scope. If it is a basic named export, such as in `export {foo}`, both `exported` and `local` are equivalent `Identifier` nodes; in this case an `Identifier` node representing `foo`. If it is an aliased export, such as in `export {bar as foo}`, the `exported` field is an `Identifier` node representing `foo`, and the `local` field is an `Identifier` node representing `bar`.
+
+## ExportDefaultDeclaration
+
+```js
+interface ExportDefaultDeclaration <: Node {
+    declaration: Declaration | Expression;
+}
+```
+
+An export default declaration, e.g., `export default function () {};` or `export default 1;`.
+
+## ExportBatchDeclaration
```

Comment:
I'm OK with `ExportAllDeclaration`, but keep in mind that we have a new feature coming up, which is exporting default from another module, where is that going to be allocated? It will be something like `export default from "mod"`, today the only way to do that is by doing `export {default} from "mod";`. I wonder if that will belong to `ExportBatchDeclaration` or `ExportDefaultDeclaration` since it is technically a batch operation with one entry to be re-exported.